pkg/activator/testing/roundtripper.go

```diff
@@ -110,9 +112,23 @@ func (rt *FakeRoundTripper) popResponse(host string) *FakeResponse {
 
 // RT is a RoundTripperFunc
 func (rt *FakeRoundTripper) RT(req *http.Request) (*http.Response, error) {
+	ctx := req.Context()
+
 	if req.Header.Get(netheader.ProbeKey) != "" {
 		rt.NumProbes.Inc()
 		resp := rt.popResponse(req.URL.Host)
+
+		// Delay if set before sending response
+		if resp.Delay.Seconds() != 0 {
+			timer := time.NewTimer(resp.Delay)
+			select {
+			case <-ctx.Done():
+				timer.Stop()
+				return nil, ctx.Err()
+			case <-timer.C:
+			}
+		}
```

Comment:
Can simplify this

```suggestion
		if resp.Delay != 0 {
			select {
			case <-ctx.Done():
				return nil, ctx.Err()
			case <-time.After(resp.Delay):
			}
		}
```

Reply:
Yeah, that is fine as well, `time.After(resp.Delay)`, from my understanding does not clean up the underlying timer if the upper case were to happen first so that timer will continue to run until it reaches the delay time.  In this particular usage having a bunch of non-needed running timers is probably both unlikely and not a huge concern.

Reply:
ah good point